CHANGELOG.md

```diff
@@ -19,6 +19,28 @@
 
 ### Compiler
 
+- The compiler now allows deprecating specific constructors of a custom type using the `@deprecated` attribute:
+
+  ```gleam
+  // Updated `Number` to `RealNumber`s only
+  pub type RealNumber {
+    @deprecated("`I` am not a real number")
+    I
+    One
```

Comment:
Can we give these some other names? `I` is a bit hard to read. Perhaps we can use the name of hash algorithms and deprecate md5?